src/Features/ERDDAP/waterLevel/timeframeSelector.tsx

```diff
@@ -16,8 +23,8 @@ export const TimeframeSelector = ({ graphFuture }: { graphFuture: boolean }) =>
   const params = useParams()
   const pathname = usePathname()
   const searchParams = useSearchParams()
-  const [startTime, setStartTime] = useState<any>(searchParams.get("start"))
-  const [endTime, setEndTime] = useState<any>(searchParams.get("end"))
+  const [startTime, setStartTime] = useState<any>(searchParams.get("start") || getIsoForPicker(daysAgoRounded(2)))
+  const [endTime, setEndTime] = useState<any>(searchParams.get("end") || getIsoForPicker(daysInFuture(3)))
```

Comment:
Seeing this pattern duplicated but slightly different makes me think we might want to refactor these out into self contained hooks that capture the functionality and complexity of `useSearchParams`, `useState`, and `useEffect`, so we might just end up with `useStartDate`, `useEndDate`, and `useDatum` that are more self explanatory (and may help contain if we refactor datums into using redux/session store).

Reply:
Cool, yeah, I can wrap that into its own hook. And agreed about the unit selection. It might be we get some feedback from users soon on what they are expecting and how we can engineer it to be more intuitive.

Reply:
I'm going to do this in another branch and merge this one